### PR TITLE
make rayon optional

### PR DESCRIPTION
**Pull request by beltegeuse**
Hi, 

This is the PR for #418. I have successfully compiled the code and pass tests.

The current default behavior (using rayon)
```
cargo test
```
To test without rayon:
```
cargo test --no-default-features
```

For now, this PR does not contain documentation updates. How do you want to handle this? Because I am not sure if you're going to put it inside the README.md or inside lib.rs (or just leave it as it).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -22,6 +22,17 @@ Maximum genericity over image storages or formats, or support for higher-dimensi
 
 Full blown computer vision applications (e.g. face recognition or image registration) probably also belong elsewhere, but the line's a bit blurred here (e.g. is image in-painting an image processing task or a computer vision task?). However, worrying about how to structure the code can probably wait until we have more code to structure...
 
+# Crate Features
+
+Imageproc is built with these features enabled by default:
+
+-   `rayon` enables multithreading image processing for certain operations (i.g., geometric transformations) via [rayon](https://github.com/rayon-rs/rayon)
```
Comment (theotherphil) [inline README.md:29]:
```suggestion
-   `rayon` enables multithreading for certain operations (e.g., geometric transformations) via [rayon](https://github.com/rayon-rs/rayon)
```

File: README.md [hunk 2]
```diff
@@ -22,6 +22,17 @@ Maximum genericity over image storages or formats, or support for higher-dimensi
...
+
+Optionally, the following dependencies can be enabled:
+
+-   `property-testing` enables testing (not that it is only for internal usage for now) with [quickcheck](https://github.com/BurntSushi/quickcheck)
```
Comment (theotherphil) [inline README.md:33]:
```suggestion
-   `property-testing` exposes helper types and methods to enable property testing via [quickcheck](https://github.com/BurntSushi/quickcheck).
```

File: README.md [hunk 3]
```diff
@@ -22,6 +22,17 @@ Maximum genericity over image storages or formats, or support for higher-dimensi
...
+-   `display-window` enables to display images (using `imageproc::window`) with [sdl2](https://github.com/Rust-SDL2/rust-sdl2)
```
Comment (theotherphil) [inline README.md:34]:
```suggestion
-   `display-window` enables the displaying of images (using `imageproc::window`) with [sdl2](https://github.com/Rust-SDL2/rust-sdl2)
```

**Comment by beltegeuse** [comment]
Thanks for reviewing the README changes. I have applied your changes and fix the style check with `cargo fmt`. I let you merge and close the related issue.

**Comment by theotherphil** [comment]
Thanks!

**Pull request merged**